traitsui/testing/tester/wx/implementation/text_editor.py

```diff
@@ -0,0 +1,86 @@
+#  Copyright (c) 2005-2020, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+#
+
+import wx
+
+from traitsui.wx.text_editor import CustomEditor, ReadonlyEditor, SimpleEditor
+from traitsui.testing.tester import command, query
+from traitsui.testing.tester.wx import helpers
+
+
+def readonly_displayed_text_handler(wrapper, interaction):
+    ''' Handler for ReadonlyEditor to handle query.DisplayedText interactions.
+
+    Parameters
+    ----------
+    wrapper : UIWrapper
+        the UIWrapper object wrapping the ReadonlyEditor
+    interaction : Instance of query.DisplayedText
+        This arguiment is not used by this function. It is included so that
+        the function matches the standard format for a handler.  The intended
+        interaction should always be query.DisplayedText
+
+    Notes
+    -----
+    wx Readonly Editors occassionally use wx.TextCtrl as their control, and
+    other times use wx.StaticText.
+    '''
+    if isinstance(wrapper.target.control, wx.TextCtrl):
+        return wrapper.target.control.GetValue()
+    elif isinstance(wrapper.target.control, wx.StaticText):
+        return wrapper.target.control.GetLabel()
+    raise TypeError("readonly_displayed_text_handler expected a UIWrapper with"
+                    " a ReadonlyEditor as a target. ReadonlyEditor control"
+                    " should always be either wx.TextCtrl, or wx.StaticText." 
+                    " {} was found".format(wrapper.target.control))
```

Comment:
This raise was a bit awkward because if someone passes a wrapper that does not have a target with a control, three would already be an error.  Maybe this should be rewritten to not take a wrapper, and instead take just a wx object, more like what was done for the qt similar function.  However, unlike with qt this wouldn't really be reusable, and in this way it can be used directly as the handler which was nice.  Any suggestions on how to raise a better error or if this is okay?

Reply:
Thank you. I think this error is okay in this context: This exception is for capturing programming error on our part. It is better to fail early than doing something unexpected silently (returning None).